RockLib.Logging/DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -204,8 +204,9 @@ private static void SetLoggerLookupDescriptor(this IServiceCollection services,
         }
 
         // Capture which loggers are singleton according to index.
-        var isSingletonLoggers = services.Where(service => service.ServiceType == typeof(ILogger))
-            .Select(service => service.Lifetime == ServiceLifetime.Singleton)
+        var singletonLoggers = services.Where(service => service.ServiceType == typeof(ILogger)
+                                            && service.Lifetime == ServiceLifetime.Singleton)
+            .Select(service => service.ImplementationInstance as ILogger)
```

Comment:
There's a `Cast<>()` method in LINQ that I'd use to do this.

Reply:
Well....maybe. You still need to do a `Select()` to grab `ImplementationInstance`.

Reply:
I think you can keep what you have, but make it a direct cast. You've already verified that it derives from `ILogger`, so I'd change it to `(ILogger)service.ImplementationInstance`. Also, you may want to use `IsAssignableFrom()` (or `IsAssignableTo()`, take your pic), instead of a strict `==`.